crates/net/network/src/transactions/fetcher.rs

```diff
@@ -357,13 +357,15 @@ impl TransactionFetcher {
         // the peer as fallback peer so it isn't request again for these hashes.
         hashes.retain(|hash| {
             if let Some(entry) = self.hashes_fetch_inflight_and_pending_fetch.get(hash) {
-                entry.fallback_peers_mut().remove(peer_failed_to_serve);
-                return true
+                entry.fallback_peers_mut().remove(&TxSizeMetadata {
+                    peer_id: *peer_failed_to_serve,
+                    tx_encoded_len: 0,
+                });
```

Comment:
this doesn't seem correct (i see this over the rest of the pr too), will this not only remove it if the entry matches the peer id and `tx_encoded_len` is 0?

Reply:
nope, because of `PartialEq` impl, it only considers the `PeerId`. at best this would be abstracted away by adding a method to `TxFetchMetadata`
```rust
pub fn remove_fallback_peer(&mut self, peer_id: PeerId) {
    ...
}
```